### DISCUSSION TITLE
How to compute the color transformation matrix for Raw RGB to known XYZ?

### DISCUSSION THREAD

**Discussion by ssh4net** [colour-science/colour#896] [Q&A] [answered]
Hi Colour wizards.

If i have measured custom color chart colors in D50 light and have Spectral, CIE LAB and XYZ values (reference values).

How can i compute color transformation matrix (3x3) from Raw RGB values i got from raw image opened with libraw as raw/linear (sensor Unknown RGB color space) to known Spectral or XYZ or CIE LAB values?

Is this possible to do such task in colour-science library?

If i remember this will required to find such matrix using many Raw RGB <-> Measured XYZ pairs to minimize matrix error
(if anyone can point me to any article about this that will be great, i seen one before but lost link)

**Comment by thomasmansencal** [comment, marked as answer]
Hi ssh4net,

Long story short: Yes this is absolutely possible, the simplest way and assuming you have a pair of data, i.e. reference and test samples, you can use one of the definitions in the [colour.characterisation.correction](https://github.com/colour-science/colour/blob/3cd6ab8d4c3483bcdeb2d7ef33967160808c0bb2/colour/characterisation/correction.py) module. Typically, a linear transformation, i.e. 1st order polynomial is used because it is exposure invariant. It is possible to use the Finlayson (2015) method with the root polynomials for same effect and then benefit from the precision of higher order/degree polynomial. The issue you might face is that polynomials tend to behave badly outside the "training" domain.

If you are looking at more refined optimisation processes, please look at the [colour.characterisation.aces_it](https://github.com/colour-science/colour/blob/3cd6ab8d4c3483bcdeb2d7ef33967160808c0bb2/colour/characterisation/aces_it.py) module.

Cheers,

Thomas
